pages/AzureTour002.md

```diff
@@ -132,62 +131,41 @@ In the following code you will invoke the <code>board.on()</code> function which
-    });
-    
-    // The htu21d.on("data", callback) function invokes the ananymous callback function at the 
-    // frequency specified in the constructor (25ms by default). The anonymous function 
-    // is scoped to the object (e.g. this == the htu21d Multi class object). 
-    htu21d.on("data", function() {
-      hF = this.temperature.fahrenheit;
-      hC = this.temperature.celsius;
-      relativeHumidity = this.hygrometer.relativeHumidity;
-      
-      // The MPL311A2 (barometer) sensor will update the global variables associated with it.
-      // Each time the HTU21D (temperature) sensor invokes this function the message sent to Azure
-      // IoT Hub will include the data from the MPL311A2 sensor as well.
-      // Create a JSON payload for the message that will be sent to Azure IoT Hub
-      var payload = JSON.stringify({
-        deviceId: deviceName,
-        location: location,
-        fahrenheit: (hF + bF) /2,
-        celsius: (hC + bC) / 2,
-        relativeHumidity: relativeHumidity,
-        pressure: pressure,
-        altitude_f: altitude_f,
-        altitude_m: altitude_m
-      });
-      
-      // Create the message based on the payload JSON
-      var message = new device.Message(payload);
-      // For debugging purposes, write out the message paylod to the console
-      console.log("Sending message: " + message.getData());
-      // Send the message to Azure IoT Hub
-      client.sendEvent(message, printResultFor('send'));
+  console.log("Board connected...");
+  // The SparkFun Weather Shield for the Particle Photon has two sensors on the I2C bus - 
+  // a humidity sensor (HTU21D) which can provide both humidity and temperature, and a 
+  // barometer (MPL3115A2) which can provide both barometric pressure and humidity.
+  // Controllers for these are wrapped in the convenient `Weather` plugin class:
+  var weather = new Weather({
+    variant: "PHOTON"
+  });
+
+  var defaults = {
+    deviceId: deviceId,
+    location: location
+  };
+
+  // The weather.on("data", callback) function invokes the anonymous callback function 
+  // whenever the data from the sensor changes (no faster than every 25ms). The anonymous 
+  // function is scoped to the object (e.g. this == the instance of Weather class object). 
+  weather.on("data", function() {
+    var payload = Object.assign({}, defaults, {
+      // celsius & fahrenheit are averages taken from both sensors
+      celsius: this.celsius,
+      fahrenheit: this.fahrenheit,
+      pressure: this.pressure,
+      feet: this.feet,
+      meters: this.meters,
+      relativeHumidity: this.relativeHumidity,
     });
+
+    // Create the message based on the payload JSON
+    var message = new device.Message(payload);
+    // For debugging purposes, write out the message paylod to the console
+    console.log("Sending message: " + message.getData());
+    // Send the message to Azure IoT Hub
+    client.sendEvent(message, printResultFor('send'));
+  });
```

Comment:
Much simpler now! Easier to get to the point :) Less teaching about mechanics of event-based software design and more about reporting IoT data!